File: core/knowledge/ontology_conflict_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.evidence.contradiction_lattice_engine import build_contradiction_lattice


def detect_ontology_conflicts(edges: List[Dict[str, Any]]) -> Dict[str, Any]:
    pairs: List[List[str]] = []
    for e in edges or []:
        c = e.get("contradictions", {}) or {}
        for p in c.get("pairs", []):
            if isinstance(p, (list, tuple)) and len(p) >= 2:
                pairs.append([str(p[0]), str(p[1])])
    lattice = build_contradiction_lattice(pairs)
    return {
        "conflicts": lattice["pairs"],
        "pressure": lattice["pressure"],
        "contradiction_pressure": lattice["pressure"],
        "uncertainty": {"visible": lattice["count"] > 0},
    }

File: src/knowledge/ontologyConflictEngine.ts
/**
 * Converted from Python: core/knowledge/ontology_conflict_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { buildContradictionLattice } from "../evidence/contradictionLatticeEngine.js";

export function detectOntologyConflicts(edges: any): any {
  var pairs: any[] = [];
  var e: any;
  for (e of py.iter(py.or2(edges, () => ([])))) {
    var c: any = py.or2(py.get(e, "contradictions", {}), () => ({}));
    var p: any;
    for (p of py.iter(py.get(c, "pairs", []))) {
      if (((Array.isArray(p) || Array.isArray(p)) && (py.len(p) >= 2))) {
        py.listAppend(pairs, [py.toStr(py.at(p, 0)), py.toStr(py.at(p, 1))]);
      }
    }
  }
  var lattice: any = buildContradictionLattice(pairs);
  return {"conflicts": py.at(lattice, "pairs"), "pressure": py.at(lattice, "pressure"), "contradiction_pressure": py.at(lattice, "pressure"), "uncertainty": {"visible": (py.at(lattice, "count") > 0)}};
}
export { buildContradictionLattice };
